mesh/mesh_readers.cpp

```diff
@@ -970,67 +984,298 @@ void Mesh::ReadGmshMesh(std::istream &input, int &curved, int &read_gf)
+            15,  /* 15-node second order prism (6 nodes associated with the
+                       vertices and 9 with the edges). */
+            13,  /* 13-node second order pyramid (5 nodes associated with the
+                       vertices and 8 with the edges). */
+            9,   /* 9-node third order incomplete triangle (3 nodes associated
+                       with the vertices, 6 with the edges) */
+            10,  /* 10-node third order triangle (3 nodes associated with the
+                       vertices, 6 with the edges, 1 with the face) */
+            12,  /* 12-node fourth order incomplete triangle (3 nodes associated
+                       with the vertices, 9 with the edges) */
+            15,  /* 15-node fourth order triangle (3 nodes associated with the
+                       vertices, 9 with the edges, 3 with the face) */
+            15,  /* 15-node fifth order incomplete triangle (3 nodes associated
+                       with the vertices, 12 with the edges) */
+            21,  /* 21-node fifth order complete triangle (3 nodes associated
+                       with the vertices, 12 with the edges, 6 with the face) */
+            4,   /* 4-node third order edge (2 nodes associated with the
+                       vertices, 2 internal to the edge) */
+            5,   /* 5-node fourth order edge (2 nodes associated with the
+                       vertices, 3 internal to the edge) */
+            6,   /* 6-node fifth order edge (2 nodes associated with the
+                       vertices, 4 internal to the edge) */
+            20,  /* 20-node third order tetrahedron (4 nodes associated with the
+                       vertices, 12 with the edges, 4 with the faces) */
+            35,  /* 35-node fourth order tetrahedron (4 nodes associated with
+                       the vertices, 18 with the edges, 12 with the faces, and 1
+                       with the volume) */
+            56,  /* 56-node fifth order tetrahedron (4 nodes associated with the
+                       vertices, 24 with the edges, 24 with the faces, and 4
+                       with the volume) */
+            -1,-1, /* unsupported tetrahedral types */
+            -1,-1, /* unsupported polygonal and polyhedral types */
+            16,  /* 16-node third order quadrilateral (4 nodes associated with
+                       the vertices, 8 with the edges, 4 wth the face) */
+            25,  /* 25-node fourth order quadrilateral (4 nodes associated with
+                       the vertices, 12 with the edges, 9 wth the face) */
+            36,  /* 36-node fifth order quadrilateral (4 nodes associated with
+                       the vertices, 16 with the edges, 16 wth the face) */
+            -1,-1,-1, /* unsupported quadrilateral types */
+            28,  /* 28-node sixth order complete triangle (3 nodes associated
+                       with the vertices, 15 with the edges, 10 with the face)
+                 */
```

Comment:
Comments on row like this (and maybe in general for this array) may look better with `//` instead of `/* ... */`:

```c++
            28,  // 28-node sixth order complete triangle (3 nodes associated
                 //    with the vertices, 15 with the edges, 10 with the face)
```

Reply:
I reformatted this section a bit -- let me know if you want me to revert it...

Reply:
Either format is fine with me.  I'll be happy to change it if you like.